src/python/pants/backend/python/goals/pytest_runner.py

```diff
@@ -381,6 +393,49 @@ async def debug_python_test(field_set: PythonTestFieldSet) -> TestDebugRequest:
     )
 
 
+@rule(desc="Set up debugpy to run an interactive Pytest session", level=LogLevel.DEBUG)
+async def debugpy_python_test(
+    field_set: PythonTestFieldSet,
+    debugpy: DebugPy,
+    pytest: PyTest,
+) -> TestDebugAdaptorRequest:
+    debugpy_pex = await Get(Pex, PexRequest, debugpy.to_pex_request())
+    if pytest.main.spec != "pytest":
+        logger.warn(
+            softwrap(
+                """
+                Ignoring custom [pytest].console_script/entry_point when using --use-debug-adaptor.
+                `debugpy` (Python's Debug Adaptor) doesn't support this use-case yet.
+                """
+            )
+        )
+
+    setup = await Get(
+        TestSetup,
+        TestSetupRequest(
+            field_set,
+            is_debug=True,
+            main=debugpy.main,
+            prepend_argv=(
+                "--listen",
+                f"{debugpy.host}:{debugpy.port}",
+                "--wait-for-client",
```

Comment:
Alternatively, we could not have the enforcement all implementations want this, and instead make the user specify `--debugpy-args=...` but this is much nicer to the user.